pinot-clients/pinot-java-client/src/main/java/org/apache/pinot/client/ResultSetStats.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.client;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+/**
+ * Simple POJO to hold query execution statistics for a request. These stats come in every
+ * query that's executed and can be used for debugging Pinot slow queries.
+ */
+class ResultSetStats {
+  private final int _numServersQueried;
+  private final int _numServersResponded;
+  private final long _numDocsScanned;
+  private final long _numEntriesScannedInFilter;
+  private final long _numEntriesScannedPostFilter;
+  private final long _numSegmentsQueried;
+  private final long _numSegmentsProcessed;
+  private final long _numSegmentsMatched;
+  private final long _numConsumingSegmentsQueried;
+  private final long _minConsumingFreshnessTimeMs;
+  private final long _totalDocs;
+  private final boolean _numGroupsLimitReached;
+  private final long _timeUsedMs;
+
+  ResultSetStats(JsonNode brokerResponse) {
+    _numServersQueried = brokerResponse.has("numServersQueried") ?
+        brokerResponse.get("numServersQueried").asInt() : -1;
```

Comment:
can we simply maintain a reference to jsonNode and look it up in the respective methods. This will add perf overhead for high throughput use cases where the client code is not interested in all the stats

Reply:
Done